### PR TITLE
Add logger to Clean command

### PR DESCRIPTION
**Pull request by jesusbv**
Add logger to `Clean` command

### CODE REVIEW

File: spec/ruby_terraform/commands/clean_spec.rb [hunk 1]
```diff
@@ -1,11 +1,20 @@
 require 'spec_helper'
 
 describe RubyTerraform::Commands::Clean do
+  before(:each) do
+    RubyTerraform.configure do |config|
+      @logger_mock = config.logger
+    end
+  end
+
```
Comment (tobyclemson) [inline spec/ruby_terraform/commands/clean_spec.rb]: To maintain test isolation, there should really be a `RubyTerraform.reset!` in an after each block.

Comment (jesusbv) [reply spec/ruby_terraform/commands/clean_spec.rb]: True, good catch. I forgot.

Comment (jesusbv) [reply spec/ruby_terraform/commands/clean_spec.rb]: Fixed

**Comment by tobyclemson** [comment]
Hey Jesus,

Happy to accept this as is but a couple of notes for any future PRs:

* I typically prefer to keep as much as possible inline in the tests rather than accessing instance variables that may bleed across test boundaries. In this case, I feel it would be better to instantiate some sort of mock logger in each test and configure it on to RubyTerraform on a per test basis. If the setup became too much I'd prefer to pull out a method to make it clear what's being set up rather than using before each. That way you can see from looking only at the test what is in scope / configured and it's easier to move tests around in future.
* Rather than capturing the logger from the config, I think it would be better to instantiate something in the test that meets the logger interface and completely replace the configured logger. Right now, if the code under test calls a method on the logger that isn't mocked, the real implementation will be invoked leading to noise in the test output. See what I did in the configuration tests for an alternative approach.

Thanks!
Toby

**Comment by tobyclemson** [comment]
Version [0.36.0](https://rubygems.org/gems/ruby-terraform/versions/0.36.0) includes this change.

Thanks,
Toby

**Pull request merged**